src/main/java/seedu/address/model/person/insights/PersonInsight.java

```diff
@@ -0,0 +1,55 @@
+package seedu.address.model.person.insights;
+
+import static seedu.address.commons.util.CollectionUtil.requireAllNonNull;
+
+import seedu.address.model.Model;
+import seedu.address.model.person.Person;
+
+
+/**
+ * This class encapsulates insights of a person.
+ */
+public class PersonInsight implements Comparable<PersonInsight> {
+
+    // Data fields
+    private final Person person;
+    private final NumLogsInsight numberOfLogs;
+    private final NumEventsInsight numberOfEvents;
+    private final MostRecentEventInsight lastEvent;
+
+    /**
+     * Constructs a PersonInsight object.
+     */
+    public PersonInsight(Person person, Model model) {
+        requireAllNonNull(person, model);
+        this.person = person;
+        this.numberOfEvents = NumEventsInsight.of(person, model);
+        this.numberOfLogs = NumLogsInsight.of(person, model);
+        this.lastEvent = MostRecentEventInsight.of(person, model);
+
+    }
+
+    public String getNumLogsInsightAsString() {
+        return this.numberOfLogs.getAsString();
+    }
+
+    public String getNumEventsInsightAsString() {
+        return this.numberOfEvents.getAsString();
+    }
+
+    public String getLastEventInsightAsString() {
+        return this.lastEvent.getAsString();
+    }
+
+    public Person getPerson() {
+        return this.person;
+    }
+
+    @Override
+    public int compareTo(PersonInsight other) {
```

Comment:
I'm guessing the + is intended behaviour? Seems like it would be a little hard to test though, but I guess it's not a big issue.

Reply:
Yep. Done for simplicity, since "insights" ranking is arbitrary. This has the benefit of extending easily for more insights, if we want.